### ISSUE TITLE
Optionally send metrics to Carbon (Graphite)

### ISSUE THREAD

**Issue by xbglowx** [ifwe/netscaler-tool#10]
It would be great if the tool could optionally send metrics to a carbon host.

**Comment by jedblack** [comment]
Any plans to have this implemented?

**Comment by xbglowx** [comment]
Hi @jedblack, I currently don't have any plans for this yet. The reason I came up with this enhancement was because I found myself writing wrapper scripts that would do the carbon submit.

I don't think it will take much time to implement. I am assuming you want this added to the 'stat' sub-command?

**Comment by jedblack** [comment]
yep, essentially, I would love to be able to pump any stat netscaler-tool can output into graphite(carbon)

**Comment by xbglowx** [comment]
Gotcha. I will try to make some time to work on this either this week or next. I will let you know when I have something, so that you can test it out.

**Comment by xbglowx** [comment]
I finally got around to coding this feature. I should have something ready by Monday.

**Comment by jedblack** [comment]
Wow, thanks Brian.  Really appreciate you taking the time.  It will be a super useful addition in my world, as I'm sure for many others as well.

**Comment by xbglowx** [comment]
@jedblack I have created a carbon branch. Can you please try it out?

1. create virtualenv
2. activate virtualenv
3. cd <git_clone_dir>
4. python setup.py install
5. modify '/etc/netscalertool.conf' and include the following at the end of the file:
carbon:
  host: host.domain.com
  port: 2003
  metric_base: foo.bar
6. netscaler-tool --carbon <netscaler_ip> stat lb-vservers totalpktssent

In the above example, the tool will send foo.bar.<netscaler_ip>.lbvserver.<lbvserver>.totalpktssent.<value> to host.domain.com:2003, for all lbvservers.

You can also try:

netscaler-tool --carbon <netscaler_ip> stat ns
netscaler-tool --carbon <netscaler_ip>  show system 

Watch /var/log/netscaler-tool/netscaler-tool.log for anything interesting and verify if metrics are making it to your carbon instance.

**Comment by jedblack** [comment]
I'm going to give it a run through the week and report back.  Thanks Brian!

**Comment by calebcall** [comment]
What value is the carbon: parameter supposed to reflect?  host:, port:, and metric_base: are all self explanatory. Thanks
